### DISCUSSION TITLE
WebRTC's transport layer, UDP or TCP?

### DISCUSSION THREAD

**Discussion by GyudongPark** [bluenviron/mediamtx#3652] [Questions]
### Question

I tried to test WebRTC's transport layer for both UDP and TCP by choosing between webrtcLocalTCPAddress and ebrtcLocalUDPAddress.

UDP is often known to be faster than TCP. But when I tested both cases, it seems that UDP is much slower than TCP if time goes on and on.

When I searched for some reasons, UDP can be slower than TCP if network buffer cannot handle the incoming packets' speed. Does WebRTC try to handle these loss or delayd UDP packets? And that makes more delay than TCP?

**Comment by iconoclasthero** [comment]
Is it supposed to buffer or just omit dropped packets and reduce quality?

**Reply by GyudongPark** [reply]
I don't do something else. Just receive the stream from mediamtx's webrtc protocol

**Comment by aler9** [comment]
Hello, when using WebRTC and the UDP transport protocol is in use, the server provides the NACK mechanism, that allows to retransmit lost packets. No delay is introduced at server side at all, with or without data losses, with or without the NACK mechanism.
Delay is eventually introduced by your player (a browser? which browser?) or by your stream source (which one?).
